modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/gossip/GossipRoundRunner.scala

```diff
@@ -51,19 +53,34 @@ object GossipRoundRunner {
         } >> S.supervise(selectPeers.foreverM).void
 
         private def evalRound(peer: Peer): F[Unit] =
-          MonadCancel[F].guarantee(
-            Temporal[F]
-              .timed(round(peer))
-              .flatMap {
-                case (duration, _) => metrics.recordRoundDuration(duration, roundLabel)
-              }
-              .flatMap(_ => metrics.incrementGossipRoundSucceeded)
-              .handleErrorWith { err =>
-                logger.error(s"Error running gossip round {peer=${peer.show}, reason=${err.show}") >>
-                  localHealthcheck.start(peer)
-              },
-            selectedPeersR.update(_.excl(peer))
-          )
```

Comment:
This is a bit overkill in here, I think this function actually is fixed correctly due to other changes, but useful information here.